### ISSUE TITLE
Confusion around `map_partitions`

### ISSUE THREAD

**Issue by ian-r-rose** [coiled/dask-community#161]
I've been seeing a lot of users confused about Dask DataFrame's partitioning and its access through `map_partitions`. (see, e.g. #151). Their questions are actually often *pandas* questions, and they seem not to understand that the argument of the `map_partitions` callback is a regular ol' pandas DataFrame.

Sometimes the answer is actually to use higher level APIs, sometimes to rewrite the function to correctly use the pandas APIs. I think part of the confusion here is that Dask DataFrame's API is so similar to that of pandas DataFrame that people who don't have a strong grasp of the implementation don't understand the difference.

This is likely solvable via more docs/tutorials. What do you think about this @pavithraes?

**Comment by pavithraes** [comment]
I agree, this does cause confusion. I think [improving the docs ](https://docs.dask.org/en/stable/generated/dask.dataframe.map_partitions.html) to make it explicit that they're working with pandas and adding an example there will go a long way!

This was the main intention behind https://docs.dask.org/en/stable/generated/dask.dataframe.map_partitions.html, but the scope of that blog increased
